src/assignment/assignment.helper.tsx

```diff
@@ -172,3 +175,175 @@ export const renderCommonMetadata = (assignment: Assignment_v2_With_Blocks): Rea
+			!assignment.description ||
+			assignment.description.length <= MAX_SEARCH_DESCRIPTION_LENGTH,
+	},
+	{
+		error: tText('assignment/assignment___de-lange-beschrijving-van-de-opdracht-is-te-lang'),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) =>
+			!(assignment as any).description_long ||
+			stripHtml((assignment as any).description_long).length <= MAX_LONG_DESCRIPTION_LENGTH,
+	},
+];
+
+const VALIDATION_RULES_FOR_PUBLISH: ValidationRule<Partial<Avo.Assignment.Assignment>>[] = [
+	{
+		error: tText('assignment/assignment___de-opdracht-heeft-geen-titel'),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) => !!assignment.title,
+	},
+	{
+		error: tText('assignment/assignment___de-opdracht-heeft-geen-beschrijving'),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) => !!assignment.description,
+	},
+	{
+		error: tText('assignment/assignment___de-opdracht-bevat-geen-onderwijsniveaus'),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) =>
+			// !assignment.loms ||
+			validateLoms(assignment?.loms, 'educationLevels'),
+	},
+	{
+		error: tText('assignment/assignment___de-opdracht-heeft-geen-vakken'),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) =>
+			validateLoms(assignment?.loms, 'subjects'),
+	},
+	{
+		error: tText('assignment/assignment___de-opdracht-heeft-geen-themas'),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) =>
+			validateLoms(assignment?.loms, 'themes'),
+	},
+	{
+		error: tText(
+			'assignment/assignment___de-tekst-items-moeten-een-titel-of-beschrijving-bevatten'
+		),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) =>
+			!assignment.blocks || validateBlocks(assignment.blocks, 'TEXT'),
+	},
+	{
+		error: tText('assignment/assignment___de-collecties-moeten-een-titel-hebben'),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) =>
+			!assignment.blocks || validateBlocks(assignment.blocks, 'COLLECTION'),
+	},
+	{
+		error: tText(
+			'assignment/assignment___de-video-items-moeten-een-titel-en-beschrijving-bevatten'
+		),
+		isValid: (assignment: Partial<Avo.Assignment.Assignment>) =>
+			!assignment.blocks || validateBlocks(assignment.blocks, 'ITEM'),
+	},
+];
+
+const validateLoms = (loms: Avo.Lom.Lom[] | undefined, scheme: LomSchemeType) => {
+	if (!loms) {
+		return false;
+	}
+
+	const lomFields = map(loms, 'lom');
+	const groupedLoms = groupLoms(lomFields);
+
+	return !!groupedLoms[scheme]?.length;
+};
+
+const validateBlocks = (blocks: Avo.Assignment.Block[], type: BlockItemTypeSchema): boolean => {
+	if (!blocks || !blocks.length) {
+		return false;
+	}
+
+	const blocksByType = blocks.filter((block) => block.type === type);
+
+	let isValid = true;
+
+	switch (type) {
+		case 'BOUW':
+		case 'COLLECTION':
+			blocksByType.forEach((block) => {
+				if (block.use_custom_fields && !block.custom_title) {
+					isValid = false;
+				}
+			});
+			break;
+		case 'ITEM':
```

Comment:
I think we have enums for these cases

Reply:
Type has type of BlockItemTypeSchema so it can only be one of those values, and VScode auto suggests those values.